internal/security/setup/option/cache_test.go

```diff
@@ -158,51 +157,42 @@ func setupCacheTest(t *testing.T) func() {
 
 	_ = setup.Init()
 
-	origScratchDir := pkiInitScratchDir
-	testScratchDir, tempDirErr := ioutil.TempDir(curDir, "scratch")
-	if tempDirErr != nil {
-		t.Fatalf("cannot create temporary scratch directory for the test: %v", tempDirErr)
-	}
-	pkiInitScratchDir = filepath.Base(testScratchDir)
-
-	origGeneratedDir := pkiInitGeneratedDir
-	testGeneratedDir, tempDirErr := ioutil.TempDir(curDir, "generated")
-	if tempDirErr != nil {
-		t.Fatalf("cannot create temporary generated directory for the test: %v", tempDirErr)
-	}
-	pkiInitGeneratedDir = filepath.Base(testGeneratedDir)
-
 	origEnvXdgRuntimeDir := os.Getenv(envXdgRuntimeDir)
 	// change it to the current working directory
 	os.Setenv(envXdgRuntimeDir, curDir)
 
-	origEnvPkiCache := os.Getenv(envPkiCache)
-	// use curDir/cache as the working directory for test
-	pkiCacheDir := filepath.Join(curDir, "cache")
-	os.Setenv(envPkiCache, pkiCacheDir)
-	_ = createDirectoryIfNotExists(pkiCacheDir)
+	testWorkDir := getWorkDir()
+	testScratchDir := filepath.Join(testWorkDir, pkiInitScratchDir)
+	if err := createDirectoryIfNotExists(testScratchDir); err != nil {
+		t.Fatalf("cannot create resource dir %s for the test: %v", testScratchDir, err)
+	}
 
-	origDeployDir := pkiInitDeployDir
-	tempDir, tempDirErr := ioutil.TempDir(curDir, "deploytest")
-	if tempDirErr != nil {
-		t.Fatalf("cannot create temporary scratch directory for the test: %v", tempDirErr)
+	testGeneratedDir := filepath.Join(testWorkDir, pkiInitGeneratedDir)
+	if err := createDirectoryIfNotExists(testGeneratedDir); err != nil {
+		t.Fatalf("cannot create resource dir %s for the test: %v", testGeneratedDir, err)
+	}
+
+	pkiCacheDir := getCacheDir()
+	if err := createDirectoryIfNotExists(pkiCacheDir); err != nil {
+		t.Fatalf("cannot create dir %s for the test: %v", pkiCacheDir, err)
+	}
+
+	pkiInitDeployDir := getDeployDir()
+	if err := createDirectoryIfNotExists(pkiInitDeployDir); err != nil {
+		t.Fatalf("cannot create dir %s for the test: %v", pkiInitDeployDir, err)
 	}
-	pkiInitDeployDir = tempDir
 
 	return func() {
 		// cleanup
 		os.Remove(jsonVaultFile)
 		os.Remove(jsonKongFile)
 		os.Setenv(envXdgRuntimeDir, origEnvXdgRuntimeDir)
-		os.Setenv(envPkiCache, origEnvPkiCache)
-		os.RemoveAll(pkiInitDeployDir)
+		os.RemoveAll(filepath.Join(curDir, "edgex")) // the base folder for testWorkDir
```

Comment:
where does this testWork Dir having "edgex" directory come from?

Reply:
it comes from the pkiInitBaseDir ("/edgex/security-secrets-setup") that's appended to the path returned by getWorkDir()... https://github.com/eno-intel/edgex-go/pull/258/files#diff-80706098e1a0d0df9d3d085e852f8586R219

Reply:
ok, but why can't just remove the `testWorkDir` created above without manipulation of joining directories?